src/algorithms/sorting/sort_struct_natural.md

```diff
@@ -0,0 +1,70 @@
+## Sort a Vector of Structs by natural ordering
+
+[![std-badge]][std] [![cat-science-badge]][cat-science]
+
+Sorts a Vector of Person structs with a single property `name` by its natural
+order (By name). In order to make Person sortable you need four traits [`Eq`],
+[`PartialEq`], [`Ord`] and [`PartialOrd`]. Eq can be simply derived.
```

Comment:
I feel that the proposed examples are quite verbose for relatively trivial functionality that is shown.

I'd at least suggest to combine the last two examples  src/algorithms/sorting/sort_struct_custom.md and  src/algorithms/sorting/sort_struct_natural.md

Also these last two examples are more focused in actually making the values sortable as opposed to actual sorting. This is not bad in itself but I fell that title does not does the example justice as the actual sorting seams to be the least important aspect of the example 

I'd also mention that `Eq`, `PartialEq`, `Ord` and `PartialOrd` can be just `#[derive]`'d. Moreover I'd only do the `impl`'s by hand if these do something interesting justifying the human work otherwise `#[derive]` is always a better / more idiomatic choice (which we should prefer), So to summarize I'd either use auto derives or make the struct more interesting to justify the manual impls.

Reply:
Hello, thanks for your suggestions. I merged the struct sorting examples and derived the structs.